smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpFileUploadManager.java

```diff
@@ -499,6 +499,9 @@ private void upload(InputStream iStream, long fileSize, Slot slot, UploadProgres
                                 + putUrl);
             }
         }
+        catch (IOException e) {
+            LOGGER.log(Level.WARNING, String.format("Exception occurred while uploading to %s", putUrl.toString()));
```

Comment:
That would swallow the `IOException` and not throw it up to the caller. That does not appear to be sensible behavior for an library.

Reply:
You are right. Changed it to a simple catch rethrow, just adding the logging. I Amended the commit

Reply:
It is unusual to log an exception and throw it. I assume your motivation is to record `putUrl`, right? That's usually the callers job to log this when he catches the exception. Another option would be to add a specialized subclass of `IOException` that also contains the `putUrl`, say `HttpUploadIoExcetpion`. But I thing that would clutter the API with next to now benefit. But maybe I am misunderstand the problem you are trying to solve here?

Reply:
I just noticed that `putUrl` is not known to the caller. In that case, having an specialized exception `HttpUploadException` appears sensible. The exception should include `slot` and `fileSize`, as I think that is all contextual information that is sensible to include.

Reply:
Indeed, that was the motivation, since putUrl is never known to the caller in some setups it can be hard to debug what server we should investigate if the upload fails. 

I added the sensible custom exception., instead of rethrowing. Let me know what you think, cheers.